src/core/data.jl

```diff
@@ -2526,16 +2526,34 @@ function calc_connected_components(data::Dict{String,<:Any}; edges=["branch", "d
         end
     end
 
-    component_lookup = Dict(i => Set{Int}([i]) for i in active_bus_ids)
-    touched = Set{Int}()
-
-    for i in active_bus_ids
-        if !(i in touched)
-            _cc_dfs(i, neighbors, component_lookup, touched)
+    sorted_bus_ids = sort(collect(active_bus_ids))
+    i0 = sorted_bus_ids[1] - 1  # this is to track un-visited buses
+    # The two dictionaries below are used to track the connected components
+    # `component_lookup` maps each bus to the ID of the connected component it belongs to, which is the smallest bus ID in said components
+    # `components` maps the connected component ID to a `Set{Int}` of all bus IDs in that component
+    component_lookup = Dict(i => i0 for i in active_bus_ids)
+    components = Dict{Int,Set{Int}}()
+
+    # ⚠️ it is important to iterate over _sorted_ bus IDs to ensure that components are labeled correctly  
+    for i in sorted_bus_ids
+        component_lookup[i] != i0 && continue  # bus already flagged; skip
```

Comment:
Can you remind me how this Julia `&&` syntax works?  I _think_ it means, if the lhs is true then execute the rhs?

Reply:
This is equivalent to
```jl
if component_lookup[i] != i0
    continue
end
```

Julia evaluates boolean expressions lazily. Thereby, given `A && B`, if `A` evaluates to `false`, then `B` is not evaluated.

Reply:
My rule is that you should always write out the `if end` version. It's much simpler to read.

Reply:
That's my preference as well, if you don't mind @mtanneau.